sdk/src/client/api/block_builder/input_selection/remainder.rs

```diff
@@ -184,30 +190,62 @@ impl InputSelection {
             return Err(Error::MissingInputWithEd25519Address);
         };
 
-        let mut remainder_builder = BasicOutputBuilder::new_with_amount(amount_diff).with_mana(mana_diff);
+        let mut remainder_outputs = Vec::new();
```

Comment:
It might be worth for readability to factor out the below code into a method called, e.g. `calculate_remainder_outputs` with the happy path (no native tokens) being at the top of method body.